From 2e8b162f7239fe46e24ab3d5b9c4c027aa8133c0 Mon Sep 17 00:00:00 2001
From: Xiongfeng Wang <wangxiongfeng2@huawei.com>
Date: Fri, 29 Jan 2021 09:31:19 +0800
Subject: [PATCH 2881/4585] [Huawei] PCI: skip disabling BAR1 for HiSilicon
 SD5991 device

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: 47455

-------------------------------------------------------------------------

Product department wants to use BAR1 for performance optimization on
SD5991 device. So let's skip disabling BAR1 for this device in
'pci_quirk_hisi_fixup_bar'

Fixes: 01d178344aad ("PCI: Add quirk for HiSilicon PCI Network Processor
devices")
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: yangerkun <yangerkun@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Patchwork Links: http://patchwork.huawei.com/project/hulk5.10/list/?series=13492
Signed-off-by: Xiongfeng Wang <wangxiongfeng2@huawei.com>
---
 drivers/pci/quirks.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/pci/quirks.c b/drivers/pci/quirks.c
index 996f454fd819..87e88e8f6a2c 100644
--- a/drivers/pci/quirks.c
+++ b/drivers/pci/quirks.c
@@ -5841,10 +5841,12 @@ static void pci_quirk_hisi_fixup_bar(struct pci_dev *dev)
 		dev->device == PCIE_DEVICE_ID_HISI_5891 ||
 		dev->device == PCIE_DEVICE_ID_HISI_5892 ||
 		dev->device == PCIE_DEVICE_ID_HISI_5981 ||
-		dev->device == PCIE_DEVICE_ID_HISI_8855 ||
-		dev->device == PCIE_DEVICE_ID_HISI_5991)
+		dev->device == PCIE_DEVICE_ID_HISI_8855)
 		start = 1;
 
+	if (dev->device == PCIE_DEVICE_ID_HISI_5991)
+		start = 2;
+
 	if (dev->device == PCIE_DEVICE_ID_HISI_5896 ||
 			dev->device == PCIE_DEVICE_ID_HISI_5886)
 		start = 3;
-- 
2.17.1

